core/tests/app/test_app.py

```diff
@@ -347,7 +347,7 @@ def test_explicit_app_metadata(monkeypatch, event_loop):
 
 
 @pytest.mark.parametrize("construct", [True, False])
-def test_icon_construction(construct, event_loop):
+def test_icon_construction(app, construct, event_loop):
```

Comment:
That's not uncommon - there's bunch of other tests that rely on App being initialised at least once before they will pass. Not sure why this particular test hasn't been a problem before, though...

Reply:
Just to be clear, this test was only a problem if you only ran it; that is, `pytest -k test_icon_construction`. Otherwise, a previous test would have initialized `toga.App.app`.